### DISCUSSION TITLE
Feedback from first manual / usabilty test

### DISCUSSION THREAD

**Discussion by mattwynne** [cucumber/polyglot-release#20] [General]
I got @aurelien-reeves to try the tool out today for releasing html-formatter 19.0.1 and collected some feedback.

1. ~~the push to the release branch failed. I didn't capture the exact error at the time, but it was some kind of `error: failed to push some refs to 'orign'`. We saw later that we had a tag simply named `v` on the release commit, so that was probably it.~~ ✅ Fixed in https://github.com/cucumber/polyglot-release/pull/21
2. ~~Aurelien signs his commits manually, so we need to add a `-S` to the `git commit` calls.~~  ✅ Fixed in https://github.com/cucumber/polyglot-release/pull/29
3. if you already have local commits to `main` they will get pushed, which might cause surprises. Need to add a guard against this. ⛏ WIP in https://github.com/cucumber/polyglot-release/pull/30
4. ~~Aurelien's workflow is always to push to a `pre-release` branch first and create PR (even though this isn't what's documented) so that the GitHub CI can run on that PR. Then he merges that PR and *then* pushes to the `release` branch. This is cautious, but it's an approach we should support with this tool.~~ ✅ Fixed in https://github.com/cucumber/polyglot-release/pull/31
5. Aurelien would like to be able to review changes before they're pushed. Pushing to a `pre-release` branch would be OK, but any pushes to `main` or a `release` branch shouldn't happen without some kind of manual intervention, by default.

**Comment by mpkorstanje** [comment]
> if you already have local commits to main they will get pushed, which might cause surprises. Need to add a guard against this.

Yes. This needs a check.

> . This is cautious, but it's an approach we should support with this tool.

Pff. Way too complicated. 

The goal is to forget about all the manual work. `make release` in `cucumber-jvm` kicks of the same commit and push action and it hasn't been CI that's been a problem for years.

> Aurelien would like to be able to review changes before they're pushed. Pushing to a pre-release branch would be OK, but any pushes to main or a release branch shouldn't happen without some kind of manual intervention, by default.

I think that `--no-git-push` should be sufficient there. Once people get used to using the tool I don't expect any one to look at the exact changes anymore.

**Comment by mattwynne** [comment]
I've added commit signing here: https://github.com/cucumber/polyglot-release/pull/29
